### PR TITLE
Added the Data property to validation so the exception thrown will co…

### PR DESCRIPTION
**Pull request by sirphilliptubell**
Adding the ability to include specific validation data in the thrown exception.

### CODE REVIEW

File: src/Vogen/Util.cs [hunk 1]
```diff
@@ -31,7 +31,9 @@ public static string GenerateValidation(VoWorkItem workItem)
             return @$"var validation = {workItem.TypeToAugment.Identifier}.{workItem.ValidateMethod.Identifier.Value}(value);
             if (validation != Vogen.Validation.Ok)
             {{
-                throw new {workItem.ValidationExceptionFullName}(validation.ErrorMessage);
+                throw new {workItem.ValidationExceptionFullName}(validation.ErrorMessage) {{
+                    Data = validation.Data
```
Comment (SteveDunn) [inline src/Vogen/Util.cs:35]: This is [failing](https://github.com/SteveDunn/Vogen/pull/375/checks#step:4:2128) with `cannot be assigned to -- it is read only`

We could either have a constructor overload, over make add an `init` accessor on the property.

Comment (viceroypenguin) [reply src/Vogen/Util.cs:35]: I would also argue that `Validation.Data` should not be instantiated with a new dictionary on object construction. For the common case where consumers do not provide data, this adds an allocation that is not necessary. Instead, it should be instantiated in `WithData` when the consumer adds the first data item.

**Comment by sirphilliptubell** [comment]
Fixed the value assignment problem and the unit tests. I left the setter on the Data property as private for now, I'm not sure that people would want to set that.

**Comment by SteveDunn** [comment]
> Fixed the value assignment problem and the unit tests. I left the setter on the Data property as private for now, I'm not sure that people would want to set that.
> 
> Unit tests should pass, two still failed locally but they looked like they were due to the time zone I'm in.

Great - thanks very much for contributing! I'd really appreciate any help with the failed timezone tests - I can't get it into a state to reproduce!

**Review by SteveDunn — APPROVED** [review]
LGTM! Thanks very much for the PR - much appreciated! 👍

**Comment by sirphilliptubell** [comment]
I added a fix that I think should fix the unit test that failed due to my timezone. At least it passes on my end.

Also, is it normal that when you run build.ps1 that it takes over an hour for this to run?

```
Passed!  - Failed:     0, Passed:  9295, Skipped:     8, Total:  9303, Duration: 1 h 25 m - SnapshotTests.dll (net6.0)
...
Passed!  - Failed:     0, Passed:  9303, Skipped:     0, Total:  9303, Duration: 19 m 5 s - SnapshotTests.dll (net7.0)
```

File: tests/Vogen.Tests/InstanceGenerationTests.cs [hunk 2]
```diff
@@ -280,14 +280,20 @@ public void It_generates_a_valid_item_when_given_a_valid_DateTime_input()
             [Fact]
             public void It_generates_a_valid_item_when_given_a_valid_DateTimeOffset_input()
             {
+                var timezoneOffset = TimeZoneInfo.Local.BaseUtcOffset;
```
Comment (SteveDunn) [inline tests/Vogen.Tests/InstanceGenerationTests.cs:283]: Works great on my machine - thank you so much - it's great to be able to get these passing everywhere!

**Comment by sirphilliptubell** [comment]
Thanks again for this awesome library, just found it recently and I'll definitely be using it a lot.

**Pull request merged**